python/taichi/lang/__init__.py

```diff
@@ -211,22 +211,22 @@ def init(arch=None,
 
 def no_activate(*args):
     for v in args:
-        taichi_lang_core.no_activate(v.ptr)
+        taichi_lang_core.no_activate(v.loop_range().ptr)
 
 
 def cache_shared(*args):
     for v in args:
-        taichi_lang_core.cache(0, v.ptr)
+        taichi_lang_core.cache(0, v.loop_range().ptr)
 
 
-def cache_read_only(v):
-    taichi_lang_core.cache(1, v.ptr)
+def cache_read_only(*args):
+    for v in args:
+        taichi_lang_core.cache(1, v.loop_range().ptr)
 
 
 def assume_in_range(val, base, low, high):
```

Comment:
What is this function used for? Can we have a section `performance.rst` in `Advanced programming` explaining these cool APIs? Or they're just incomplete or even buggy for now?